omniscidb/QueryEngine/CostModel/CostModel.h

```diff
@@ -14,37 +14,56 @@
 #pragma once
 
 #include <memory>
-#include <mutex>
+#include <shared_mutex>
 
 #include "DataSources/DataSource.h"
-#include "ExtrapolationModels/ExtrapolationModel.h"
+#include "ExtrapolationModels/ExtrapolationModelProvider.h"
 #include "Measurements.h"
 
 #include "Dispatchers/ExecutionPolicy.h"
 #include "QueryEngine/CompilationOptions.h"
-#include "QueryEngine/Descriptors/RelAlgExecutionDescriptor.h"
 
 namespace costmodel {
 
 struct CaibrationConfig {
   std::vector<ExecutorDeviceType> devices;
 };
 
+struct QueryInfo {
+  std::vector<AnalyticalTemplate> templs;
+  size_t bytes_size;
+};
+
+struct CostModelConfig {
```

Comment:
Do we expect more fields here? I.e. why use a structure for a single entry?

Reply:
Yes, I would like to leave it like this, because it seems to me that new arguments may quietly appear in the future.